### PR TITLE
pkg/apis: Add support to specify relabel actions in CamelCase

### PR DESCRIPTION
**Pull request by slashpai**
Adds support to specify relabel actions in CamelCase format
to follow https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants
and at the same time always downcase the value in the generated config.
Eventually remove the lowercase way of specifying actions once
monitoring.coreos.com/v2 is created

Signed-off-by: Jayapriya Pai <slashpai9@gmail.com>

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [x] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Add support to specify relabel actions in CamelCase
```

### CODE REVIEW

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -1030,7 +1030,7 @@ type RelabelConfig struct {
 	Replacement string `json:"replacement,omitempty"`
 	//Action to perform based on regex matching. Default is 'replace'.
 	//uppercase and lowercase actions require Prometheus >= 2.36.
-	//+kubebuilder:validation:Enum=replace;keep;drop;hashmod;labelmap;labeldrop;labelkeep;uppercase;lowercase
+	//+kubebuilder:validation:Enum=replace;Replace;keep;Keep;drop;Drop;hashmod;HashMod;labelmap;LabelMap;labeldrop;LabelDrop;labelkeep;LabelKeep;lowercase;Lowercase;uppercase;Uppercase
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:1033]: Not feeling strongly but LowerCase and UpperCase look better to my eyes...

```suggestion
	//+kubebuilder:validation:Enum=replace;Replace;keep;Keep;drop;Drop;hashmod;HashMod;labelmap;LabelMap;labeldrop;LabelDrop;labelkeep;LabelKeep;lowercase;LowerCase;uppercase;UpperCase
```

**Review by simonpasquier — COMMENTED** [review]
Can you add a test in pkg/prometheus/promcfg.go?

File: pkg/apis/monitoring/v1/types.go (hunk 1 above)
Comment (slashpai) [reply pkg/apis/monitoring/v1/types.go:1033]: I thought so as well but I kept it similar to https://github.com/prometheus/prometheus/blob/0c3abdc26d4cd2e44549b122799c652c9796f8d8/model/relabel/relabel.go#L43-L59 but can update like this if thats better

Comment (simonpasquier) [reply pkg/apis/monitoring/v1/types.go:1033]: All good then, it makes more sense to be consistent with prometheus/prometheus.

**Comment by slashpai** [comment]
@simonpasquier Updated with test

**Review by simonpasquier — COMMENTED** [review]
just one more place to "fix" :)
https://github.com/prometheus-operator/prometheus-operator/blob/d4b490f6f93082e88560509fef4d8dc10e42a110/pkg/prometheus/promcfg.go#L1865

**Comment by slashpai** [comment]
Done

**Pull request merged**